rfcs/0001-multi-tenancy/README.md

```diff
@@ -0,0 +1,165 @@
+# RFC-0001 Flux Multi-Tenancy
+
+## Summary
+
+The main goal of this RFC is to define the Kubernetes tenancy models supported by Flux.
+
+## Motivation
+
+The Flux documentation makes references to account impersonation, reconciliation isolation and other multi-tenancy
+related features without clearly defining which tenancy models are supported by Flux and how they relate to source
+access control and Kubernetes role-based access control.
+
+### Goals
+
+- Define multi-tenancy from a Flux user perspective.
+- List the types of users that interact with Flux in regard to multi-tenancy.
+- List the tenancy models supported by Flux.
+- Explain the differences between tenancy models.
+
+### Non-Goals
+
+- Explain in detail the Kubernetes architecture in regard to multi-tenancy.
+- Provide an end-to-end workflow of how to setup multi-tenancy with Flux.
+
+## Proposal
+
+The Flux documentation should have a section dedicated to multi-tenancy. What follows is the proposed content
+of the multi-tenancy overview documentation that sets the stage for future enhancements.
+
+### Introduction
+
+Flux allows different organizations and/or teams to share the same Kubernetes control plane to deliver applications
+in a GitOps manner. Flux enables segmentation and isolation of resources across tenants by leveraging
+Kubernetes Cluster API, namespaces and role-based access control.
+
+### User Roles
+
+There are two types of users that interact with Flux: platform admins and tenants.
+Besides installing Flux, all the other operations (deploy applications, configure ingress, policies, etc)
+do not require users to have direct access to the Kubernetes API. Flux acts as a proxy between users and
+the Kubernetes API, using Git as source of truth for the cluster desired state. Changes to the clusters
+and workloads configuration can be made in a collaborative manner, where the various teams responsible for
+the delivery process propose, review and approve changes via pull request workflows.
+
+#### Platform Admins
+
+The platform admins have unrestricted, cluster-scoped access to Kubernetes API.
+They are responsible for installing Flux and granting Flux
+access to the sources (Git, Helm, OCI repositories) that make up the cluster(s) control plane desired state.
```

Comment:
Does this mean that tenants should not configure their own sources? In the tenants section it does however state "Register their sources with Flux". I might just be misinterpreting the meaning.

Reply:
This is about the `cluster control plane desired state` as in cluster-wide resources, controllers, etc.

Reply:
I've added a new section "Tenants Onboarding". Hopefully this clarifies that tenants can add their app repos to their main repo which is registered by admins.